### PR TITLE
nocrash.py: check git user name

### PR DESCRIPTION
**Pull request by tripleee**
Fail if wrong or unset unless running with no-git-user-check

### CODE REVIEW

File: nocrash.py [hunk 1]
```diff
@@ -59,6 +60,19 @@ def error(message):
     logging.error('[NoCrash] {}'.format(message))
 
 
+if 'no-git-user-check' in persistent_arguments:
+    persistent_arguments.remove('no-git-user-check')
+else:
+    git_name = git.config('--get', 'user.name', _ok_code=[0, 1])
+    if git_name != "SmokeDetector":
+        logging.error('git config user.name "{0}" is wrong'.format(git_name))
```
Comment (NobodyNada) [inline nocrash.py]: Maybe mention that the error can be bypassed; e.g. `git config user.name "{0}" is wrong; use no-git-check to ignore`

Comment (tripleee) [reply nocrash.py]: Yeah, I convinced myself that short source lines would be more important than useful feedback, which is obviously not a good balance.  Hang on.

**Comment by teward** [comment]
Y'know, if we're just checking statically if the credentials are set correctly, why don't we just let nocrash.py *set* the credentials if they don't exist, just for the Smokey repository?

**Comment by tripleee** [comment]
I was considering that, but permanently overriding the user's preference e.g. in a development instance would be a huge misfeature IMHO. Granteoe, you could just say don't use nocrash if you aren't running a production instance.

**Comment by ArtOfCode-** [comment]
!!/approve

**Pull request merged**